js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt

```diff
@@ -55,8 +78,14 @@ class SolanaMobileWalletAdapterModule(reactContext: ReactApplicationContext) :
                 localAssociation.port,
                 localAssociation.session
             )
-            currentActivity?.startActivityForResult(intent, 0)
-                ?: throw NullPointerException("Could not find a current activity from which to launch a local association")
+            launchIntent(intent, REQUEST_LOCAL_ASSOCIATION) { resultCode, _ ->
```

Comment:
`startActivityForResult` is deprecated.

Reply:
I'm mainly curious about the callback and how `promise` works in Kotlin. Why doesn't `line 92: promise.resolve(true)` execute before `promise.reject(...)` in the callback?

Reply:
@Ankur this is React Native. `startActivityForResult` is appropriate here, check the RN docs.

Reply:
@Michaelsulistio because of the get() call on line 90. localAssociation.start().get(...) will block until the connection is established. If the intent is cancelled by the user, the wallet will never receive the intent, so the call to localAssociation.start() will not succeed. 

Now, one thing I did not fully test is what if the user allows the timeout to expire then dismisses the intent chooser. I might need another try catch around the `promise.reject()` call on line 84 in case that promise has already gone out of scope.

Reply:
> what if the user allows the timeout to expire then dismisses the intent chooser

tested this, the dapp will first receive the  `Unable to connect to websocket server` error, as expected. If the user cancels the intent chooser after this point, the callback is still fired but nothing happens.

Reply:
> localAssociation.start().get(...) will block until the connection is established

ah thats the part I was missing. That part makes sense now! 

> the callback is still fired but nothing happens.

I just realized/remembered that the `promise` type is a RN-Kotlin Argument type and not a Native Kotlin type. This makes sense because trying to settle a fulfilled promise in JS doesn't do anything.